pkg/scheduler/scheduler.go

```diff
@@ -770,13 +723,77 @@ func (s *Scheduler) rescheduleResourceBinding(resourceBinding *workv1alpha1.Reso
 	}
 	klog.Infof("The final binding.Spec.Cluster values are: %v\n", resourceBinding.Spec.Clusters)
 
-	_, err = s.KarmadaClient.WorkV1alpha1().ResourceBindings(ns).Update(context.TODO(), resourceBinding, metav1.UpdateOptions{})
+	_, err = s.KarmadaClient.WorkV1alpha1().ResourceBindings(resourceBinding.Namespace).Update(context.TODO(), resourceBinding, metav1.UpdateOptions{})
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
+func getReservedCluster(clusters []workv1alpha1.TargetCluster, readyClusterInfos []*framework.ClusterInfo) sets.String {
+	boundClusters := util.GetClusterSetString(clusters)
+	readyClusters := getReadyCluster(readyClusterInfos)
+
+	return boundClusters.Difference(boundClusters.Difference(readyClusters))
+}
+
+func getAvailableCluster(clusters []workv1alpha1.TargetCluster, readyClusterInfos []*framework.ClusterInfo) sets.String {
+	boundClusters := util.GetClusterSetString(clusters)
+	readyClusters := getReadyCluster(readyClusterInfos)
+
+	return readyClusters.Difference(boundClusters)
+}
+
+func getReadyCluster(readyClusterInfos []*framework.ClusterInfo) sets.String {
```

Comment:
The function name is misleading.

Reply:
I move this func to `/pkg/scheduler/cache/snapshot.go`, and rename it to `GetReadyClusterNames`.